packages/compiler/src/lib/decorators.ts

```diff
@@ -164,17 +165,30 @@ export function getIndexer(program: Program, target: Type): ModelIndexer | undef
 export function isStringType(program: Program | ProjectedProgram, target: Type): target is Scalar {
   const coreType = program.checker.getStdType("string");
   const stringType = target.projector ? target.projector.projectType(coreType) : coreType;
-  return (
-    target.kind === "Scalar" && program.checker.isTypeAssignableTo(target, stringType, target)[0]
-  );
+  return isOrUnionOfOrNull(target, (type) => {
```

Comment:
I feel like updating `isStringType` to allow null values might be a bit confusing? Instead I think you could update decorators to call the new isOrUnionOfOrNull` helper which would be more explicit?

Reply:
Yeah I guess it made a bit more sense on the isNumeric where if it is a union of various numeric types it is still ok